core/src/main/scala/com/spotify/featran/transformers/Transformer.scala

```diff
@@ -70,6 +70,31 @@ abstract class Transformer[-A, B, C](val name: String) extends Serializable {
   protected def nameAt(n: Int): String = name + '_' + n
   protected def names(n: Int): Stream[String] = (0 until n).toStream.map(nameAt)
 
+  /**
+   * Builds a new transformer with an extra input preprocessing step
+   * @param f input preprocessing function
+   */
+  def contramap[AA](f: AA => A): Transformer[AA, B, C] = {
```

Comment:
```suggestion
  def contramap[AA](f: AA => A): Transformer[AA, B, C] = { self =>
```

Reply:
Addressed in 6fbfbc0

This suggestion doesn't actually compile, so I had to declare `self` on the class declaration, I hope that's OK:

```scala
abstract class Transformer[-A, B, C](val name: String) extends Serializable { self =>
```

Reply:
Right, I misread :joy: